app/src/main/java/com/asset/mvvmasset/di/RetrofitModule.kt

```diff
@@ -0,0 +1,36 @@
+package com.asset.mvvmasset.di
+
+import com.asset.mvvmasset.BuildConfig
+import com.asset.mvvmasset.data.api.MoviesApiService
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
+
+@InstallIn(SingletonComponent::class)
+@Module
+object RetrofitModule {
+
+    @Singleton
+    @Provides
+    fun provideGsonBuilder(): Gson = GsonBuilder().create()
+
+    @Singleton
+    @Provides
+    fun provideRetrofit(gson: Gson): Retrofit.Builder {
+        return Retrofit.Builder()
+            .baseUrl(BuildConfig.BASE_URL)
+            .addConverterFactory(GsonConverterFactory.create(gson))
+    }
+
+    @Singleton
+    @Provides
+    fun provideBlogApi(retrofit: Retrofit.Builder): MoviesApiService {
```

Comment:
We can have a inline function for this with a generic type for interfaces like this:

inline fun <reified T> createWebService(retrofit: Retrofit): T = retrofit.create(T::class.java)

Usage:

createWebService<MoviesApiService>(retrofit_instance)

Reply:
Yes, we can, will do and update the PR, Thanks!